### PR TITLE
Update to WASDK 1.4 and related maintenance

### PR DESCRIPTION
**Pull request by Scottj1s**
A bit of a hodgepodge PR:
Update to WASDK 1.4
Work around menu theme regression with 1.4
Eliminate WindowsAppSDK-VersionInfo.cs duplicate source warning
Remove bogus patch number
Support self-contained
Add publish profiles
Disable debug publish optimizations (inner loop)
Remove wapproj cruft

### CODE REVIEW

**Comment by Scottj1s** [comment]
See also:
https://github.com/microsoft/WindowsAppSDK/pull/3819

File: WinUIGallery/Controls/PageHeader.xaml [hunk 1]
```diff
@@ -27,7 +27,6 @@
         </Grid.RowDefinitions>
 
         <TextBlock
-            AutomationProperties.HeadingLevel="Level1"
```
Comment (Scottj1s) [inline WinUIGallery/Controls/PageHeader.xaml:30]: fix build break from yesterday:
https://github.com/microsoft/WinUI-Gallery/commit/913286a437b6ad3e4d74a493e7a08a5629807963

File: WinUIGallery/WinUIGallery.csproj [hunk 2]
```diff
@@ -17,252 +17,266 @@
...
+        <Packaged Condition="'$(Configuration)' == 'Debug-Unpackaged' Or '$(Configuration)' == 'Release-Unpackaged'">false</Packaged>
+        <Optimized>true</Optimized>
+        <Optimized Condition="'$(Configuration)' == 'Debug-Unpackaged' Or '$(Configuration)' == 'Debug'">false</Optimized>
+        <!--<WindowsAppSdkSelfContained>true</WindowsAppSdkSelfContained>-->
+    </PropertyGroup>
 
-      <PropertyGroup Condition="'$(Configuration)' == 'Debug-Unpackaged' Or '$(Configuration)' == 'Release-Unpackaged'">
-          <EnableMsixTooling>false</EnableMsixTooling>
-          <WindowsPackageType>None</WindowsPackageType>
-          <WindowsAppSdkEnableMdd>true</WindowsAppSdkEnableMdd>
```
Comment (Scottj1s) [inline WinUIGallery/WinUIGallery.csproj:29]: support WindowsAppSdkSelfContained, don't assume framework-dependent

File: WinUIGallery/WinUIGallery.csproj [hunk 3]
```diff
@@ -17,252 +17,266 @@
...
-          <GenerateAppInstallerFile>true</GenerateAppInstallerFile>
-          <!-- URI must be absolute so this is really only useful for testing a locally produced
+    <PropertyGroup Condition="'$(Packaged)' != 'true'">
+        <PublishProfile>win10-$(Platform)-unpackaged.pubxml</PublishProfile>
+        <EnableMsixTooling>false</EnableMsixTooling>
+        <WindowsPackageType>None</WindowsPackageType>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(Packaged)' == 'true'">
+        <PublishProfile>win10-$(Platform).pubxml</PublishProfile>
```
Comment (Scottj1s) [inline WinUIGallery/WinUIGallery.csproj:38]: add publish profile property to support publishing

File: WinUIGallery/Properties/PublishProfiles/win10-arm64-unpackaged.pubxml [hunk 4]
```diff
@@ -9,8 +9,8 @@ https://go.microsoft.com/fwlink/?LinkID=208121.
     <RuntimeIdentifier>win10-arm64</RuntimeIdentifier>
     <SelfContained>true</SelfContained>
     <PublishSingleFile>False</PublishSingleFile>
-    <PublishReadyToRun>False</PublishReadyToRun>
     <PublishAppxPackage>False</PublishAppxPackage>
-    <PublishTrimmed>True</PublishTrimmed>
+    <PublishReadyToRun>$(Optimized)</PublishReadyToRun>
```
Comment (Scottj1s) [inline WinUIGallery/Properties/PublishProfiles/win10-arm64-unpackaged.pubxml:13]: enable publish optimizations only for release (faster debug inner loop)

File: WinUIGallery/WinUIGallery.csproj [hunk 5]
```diff
@@ -17,252 +17,266 @@
...
+    </ItemGroup>
+    <ItemGroup>
+        <Page Update="Controls\DesignGuidance\ColorSections\TextSection.xaml">
+            <Generator>MSBuild:Compile</Generator>
+        </Page>
+        <Page Update="SamplePages\SampleBuiltInSystemBackdropsWindow.xaml">
+            <XamlRuntime>$(DefaultXamlRuntime)</XamlRuntime>
+        </Page>
+    </ItemGroup>
+    <Import Project="ContentIncludes.props" />
```
Comment (karkarl) [inline WinUIGallery/WinUIGallery.csproj:270]: Curious why there are all these diffs here?

File: WinUIGallery/Controls/PageHeader.xaml (hunk 1 above)
Comment (karkarl) [reply WinUIGallery/Controls/PageHeader.xaml:30]: main should have the latest build fixes now.

**Comment by Scottj1s** [comment]
Note: build failing because in Microsoft.WindowsAppSDK.DeploymentManager.CS.targets, the target GenerateDeploymentManagerCS renamed to GenerateBootstrapCS between 1.3 and 1.4.  When 1.4 is released, I'll update this PR to use it, which should resolve the build issue.

**Comment by Scottj1s** [comment]
> > Note: build failing because in Microsoft.WindowsAppSDK.DeploymentManager.CS.targets, the target GenerateDeploymentManagerCS renamed to GenerateBootstrapCS between 1.3 and 1.4. When 1.4 is released, I'll update this PR to use it, which should resolve the build issue.
> 
> We should be sure to relnote this. We should also provide a less brittle way of doing this (or move this into our props/targets so that every app who wants the version number can get it easily).

good idea - updated the related WinAppSDK PR to simply use a property for including version info exactly once, before any compilation

**Pull request merged**